### PR TITLE
feat: DID linked domains service validation function

### PR DESCRIPTION
**Pull request by DRK3**
Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

**Review by vkubiv — APPROVED** [review]

**Review by sandrask — APPROVED** [review]

File: cmd/wallet-sdk-gomobile/did/wellknown.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+type ValidationResult struct {
+	IsValid    bool
+	ServiceURL string
+}
+
+// ValidateService validates the given DID's service against its well-known DID configuration.
+// It returns a ValidationResult.
+// The DID document must specify only a single service. If there are multiple URLs for a given service, only the
+// first will be checked.
+func ValidateService(did string, resolver api.DIDResolver) (*ValidationResult, error) {
```
Comment (rolsonquadras) [inline cmd/wallet-sdk-gomobile/did/wellknown.go:32]: Should this be names ValidateWellKnownDIDConfig as that's the actual validation? There can be multiple services, but this specific validation is for Well Known DID Config.

Comment (DRK3) [reply cmd/wallet-sdk-gomobile/did/wellknown.go:32]: I think `ValidateWellKnownDIDConfig` would be somewhat misleading, as the main point of this function isn't validating the well-known DID config itself, but rather is validating the service in the DID doc _against_ its well-known DID config. (The function in Aries that I call does happen to also validate the well-known DID config along the way, but I think that's just a piece of the overall picture)

Comment (rolsonquadras) [reply cmd/wallet-sdk-gomobile/did/wellknown.go:32]: There could be multiple services inside a DID Document. We are validating services with `"type": "LinkedDomains"` - https://identity.foundation/.well-known/resources/did-configuration/#example-7.

Comment (sudeshrshetty) [reply cmd/wallet-sdk-gomobile/did/wellknown.go:32]: agree with @rolsonquadras 
`VaidateService` is too generic name, should be called `ValidateLinkedDomain, ValidateWellKnown or ValidateWellKnownConfig`

Comment (DRK3) [reply cmd/wallet-sdk-gomobile/did/wellknown.go]: `ValidateLinkedDomains` sounds like a better name to me. I've renamed it to that (and also added in a check to ensure the service is of type `LinkedDomains`)

File: pkg/did/wellknown/wellknown.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	didDocResolution, err := resolver.Resolve(did)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to resolve DID: %w", err)
+	}
+
+	if len(didDocResolution.DIDDocument.Service) == 0 {
+		return false, "", errors.New("resolved DID document has no services specified")
+	} else if len(didDocResolution.DIDDocument.Service) > 1 {
```
Comment (rolsonquadras) [inline pkg/did/wellknown/wellknown.go:48]: This condition is not valid. The DID Document can have more than 1 services. For LinkedDomain, we said we will restrict to 1. If DID Document has 1 Linked Domain type and a different type service, this case will fail.

Comment (DRK3) [reply pkg/did/wellknown/wellknown.go]: Fixed.

**Review by rolsonquadras — APPROVED** [review]

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**